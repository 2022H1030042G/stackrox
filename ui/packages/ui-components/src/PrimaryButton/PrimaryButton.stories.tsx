import React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';

import PrimaryButton from './PrimaryButton';

export default {
    title: 'PrimaryButton',
    component: PrimaryButton,
} as Meta;

export const BasicUsage: Story = () => {
    function onClick(): void {}
    return (
        <PrimaryButton type="button" onClick={onClick}>
            Save the Planet
        </PrimaryButton>
    );
};
