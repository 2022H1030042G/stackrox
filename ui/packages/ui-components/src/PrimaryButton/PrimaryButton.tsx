import React, { ReactElement } from 'react';

import Button, { HOCButtonProps } from '../Button';

function PrimaryButton({
    type = 'button',
    onClick,
    children,
    isDisabled = false,
}: HOCButtonProps): ReactElement {
    return (
        <Button colorType="primary" type={type} onClick={onClick} isDisabled={isDisabled}>
            {children}
        </Button>
    );
}

export default PrimaryButton;
