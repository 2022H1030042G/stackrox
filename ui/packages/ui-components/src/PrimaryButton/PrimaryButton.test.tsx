import React from 'react';
import { render } from '@testing-library/react';

import PrimaryButton from './PrimaryButton';

describe('PrimaryButton', () => {
    it('renders', () => {
        function onClick(): void {}

        const { getByText } = render(
            <PrimaryButton type="button" onClick={onClick}>
                Save cluster
            </PrimaryButton>
        );

        expect(getByText('Save cluster')).toBeInTheDocument();
    });

    it('has the correct style', () => {
        function onClick(): void {}

        const { container } = render(
            <PrimaryButton type="button" onClick={onClick}>
                Click me!
            </PrimaryButton>
        );

        expect(container).toMatchSnapshot();
    });
});
