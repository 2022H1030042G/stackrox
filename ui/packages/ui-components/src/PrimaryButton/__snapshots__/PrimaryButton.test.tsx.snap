// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton has the correct style 1`] = `
<div>
  <button
    class="border-2 font-600 inline-flex items-center justify-center rounded-sm uppercase text-sm border-primary-500 bg-primary-200 hover:bg-primary-300 hover:text-primary-800 text-primary-700 p-2"
    type="button"
  >
    Click me!
  </button>
</div>
`;
